src/Symfony/Component/Security/Http/RememberMe/AbstractRememberMeHandler.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\RememberMe;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @experimental in 5.3
+ */
+abstract class AbstractRememberMeHandler implements RememberMeHandlerInterface
+{
+    private $userProvider;
+    protected $requestStack;
+    protected $options;
+    protected $logger;
+
+    public function __construct(UserProviderInterface $userProvider, RequestStack $requestStack, array $options = [], ?LoggerInterface $logger = null)
+    {
+        $this->userProvider = $userProvider;
+        $this->requestStack = $requestStack;
+        $this->options = $options + [
+            'name' => 'REMEMBERME',
+            'lifetime' => 31536000,
+            'path' => '/',
+            'domain' => null,
+            'secure' => false,
+            'httponly' => true,
+            'samesite' => null,
+            'always_remember_me' => false,
+            'remember_me_parameter' => '_remember_me',
+        ];
+        $this->logger = $logger;
+    }
+
+    /**
+     * @throws AuthenticationException
+     */
+    abstract protected function processRememberMe(RememberMeDetails $rememberMeDetails, UserInterface $user): void;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function consumeRememberMeCookie(RememberMeDetails $rememberMeDetails): UserInterface
+    {
+        try {
+            $user = $this->userProvider->loadUserByUsername($rememberMeDetails->getUsername());
+        } catch (\Exception $e) {
+            if (!$e instanceof AuthenticationException) {
```

Comment:
Throwing something else that an `AuthenticationException` means transgressing the `loadUserByUsername()`' contract. We should not silently fix such bad logic (not sure we should care at all)

Reply:
This was a copy-past from the current remember me system, but I agree. Let's fix it